jaxb/src/main/java/com/baeldung/jaxb/dateunmarshalling/DateAdapter.java

```diff
@@ -0,0 +1,25 @@
+package com.baeldung.jaxb.dateunmarshalling;
+
+import javax.xml.bind.annotation.adapters.XmlAdapter;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+public class DateAdapter extends XmlAdapter<String, Date> {
+
+    private final SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+
+    @Override
+    public String marshal(Date v) {
+        synchronized (dateFormat) {
```

Comment:
I appreciate that this is not threadsafe... would it be safer/simpler to make the `yyyy-MM-dd...` the `private static final` thing and create a new `SimpleDateFormat` object every time?